docs/velocity/dev/api/plugin-messaging.md

```diff
@@ -0,0 +1,160 @@
+---
+slug: /dev/plugin-messaging
+description: How to handle and send plugin messages on Velocity.
+---
+
+# Plugin Messaging
+
+First introduced in [2012](https://web.archive.org/web/20220711204310/https://dinnerbone.com/blog/2012/01/13/minecraft-plugin-channels-messaging/),
+Plugin messaging is a way for Velocity plugins to communicate with clients and backend servers. 
+
+Velocity manages connections in both directions,
+for both the client and backend server.
+
+This means Velocity plugins need to consider 4 main cases. 
+
+```mermaid
+flowchart LR
+    player -->|"1 (Incoming)"| velocity -->|"2 (Outgoing)"| backend
+    backend -->|"3 (Incoming)"| velocity -->|"4 (Outgoing)"| player
+```
+
+Additionally, BungeeCord channel compatability is included, which may remove the need for a companion velocity plugin certain cases.
```

Comment:
```suggestion
Additionally, BungeeCord compatibility is included, which may remove the need for a companion Velocity plugin in certain cases.
```

Reply:
i specifically included "channel" as other aspects aren't compatible (i.e plugins)

Reply:
cool beans, just apply the rest of the diff please and thanks 🙏